sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Processor/PartitionLoadBalancer.cs

```diff
@@ -436,7 +435,8 @@ private async Task RenewOwnershipAsync(CancellationToken cancellationToken)
 
             Logger.RenewOwnershipStart(OwnerIdentifier);
 
-            IEnumerable<EventProcessorPartitionOwnership> ownershipToRenew = InstanceOwnership.Values
+            List<EventProcessorPartitionOwnership> ownershipToRenew = InstanceOwnership.Values
+                .Where(ownership => (ownership.LastModifiedTime - DateTimeOffset.UtcNow) > LoadBalanceInterval)
```

Comment:
Should we consider hoisting the `now` calculation out so that it's a consistent value for each use, or do you think the small spread in elapsed time helps by avoiding clustering?

Reply:
Yep